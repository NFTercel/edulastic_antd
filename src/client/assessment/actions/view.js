import { CHANGE_PREVIEW } from "../constants/actions";

// change test item view type
export const changePreview = view => ({
  type: CHANGE_PREVIEW,
  payload: {
    view
  }
});
