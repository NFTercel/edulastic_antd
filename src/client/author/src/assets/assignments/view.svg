<svg xmlns="http://www.w3.org/2000/svg" width="15.143" height="9.65" viewBox="0 0 15.143 9.65"><g transform="translate(0 -92.837)"><g transform="translate(0 92.835)"><g transform="translate(0 0)"><path d="M15.046,97.366c-.135-.185-3.358-4.531-7.475-4.531S.231,97.181.1,97.366a.5.5,0,0,0,0,.589c.135.185,3.358,4.531,7.475,4.531s7.34-4.346,7.475-4.531A.5.5,0,0,0,15.046,97.366Zm-7.475,4.122c-3.032,0-5.658-2.884-6.436-3.827.776-.944,3.4-3.827,6.436-3.827s5.658,2.884,6.436,3.827C13.23,98.6,10.61,101.488,7.571,101.488Z" transform="translate(0 -92.835)" fill="#434b5d"/></g></g><g transform="translate(4.576 94.665)"><g transform="translate(0 0)"><path d="M157.717,154.725a3,3,0,1,0,3,3A3,3,0,0,0,157.717,154.725Z" transform="translate(-154.722 -154.725)" fill="#434b5d"/></g></g></g></svg>