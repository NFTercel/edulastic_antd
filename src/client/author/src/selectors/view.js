import { createSelector } from "reselect";

const moduleName = "view";
export const stateSelector = state => state[moduleName];

export const getViewSelector = createSelector(
  stateSelector,
  state => state.view
);

export const getPreviewSelector = createSelector(
  stateSelector,
  state => state.preview
);
