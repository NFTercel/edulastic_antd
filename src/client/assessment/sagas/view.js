import { takeEvery, all, put } from "redux-saga/effects";
import { CHANGE_PREVIEW, SET_ANSWER, CLEAR_ANSWERS, REMOVE_ANSWERS } from "../constants/actions";

function* resetView() {
  yield put({
    type: CHANGE_PREVIEW,
    payload: {
      view: "clear"
    }
  });
}

function* clearAnswers() {
  yield put({
    type: REMOVE_ANSWERS
  });

  yield put({
    type: CHANGE_PREVIEW,
    payload: {
      view: "clear"
    }
  });
}

export default function* watcherSaga() {
  yield all([yield takeEvery(SET_ANSWER, resetView), yield takeEvery(CLEAR_ANSWERS, clearAnswers)]);
}
