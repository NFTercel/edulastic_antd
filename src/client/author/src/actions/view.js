import { CHANGE_VIEW, CHANGE_PREVIEW } from "../constants/actions";

export const changeViewAction = view => ({
  type: CHANGE_VIEW,
  payload: {
    view
  }
});

// change test item view type
export const changePreviewAction = view => ({
  type: CHANGE_PREVIEW,
  payload: {
    view
  }
});

export default changeViewAction;
