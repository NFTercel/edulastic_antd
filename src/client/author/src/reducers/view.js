import { CHANGE_VIEW, CHANGE_PREVIEW } from "../constants/actions";

const initialState = {
  view: "edit",
  preview: "clear",
  showAnswers: false
};

export default function reducer(state = initialState, { type, payload }) {
  switch (type) {
    case CHANGE_VIEW:
      return { ...state, view: payload.view };
    case CHANGE_PREVIEW:
      return {
        ...state,
        preview: payload.view
      };
    default:
      return state;
  }
}
