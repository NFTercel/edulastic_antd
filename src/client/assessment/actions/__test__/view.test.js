import { changePreview } from "../view";

describe("View", () => {
  it("change preview should return an action", () => {
    expect(changePreview({ view: "check" })).toMatchSnapshot();
  });
});
