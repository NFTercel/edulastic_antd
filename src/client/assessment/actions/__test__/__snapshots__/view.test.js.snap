// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View change preview should return an action 1`] = `
Object {
  "payload": Object {
    "view": Object {
      "view": "check",
    },
  },
  "type": "[view] change view",
}
`;
